### PR TITLE
Add timeout to CI step that runs the unit tests

### PR DESCRIPTION
**Pull request by dalg24**
Related to https://github.com/arborx/ArborX/pull/163#issuecomment-546331977

### CODE REVIEW

File: .jenkins [hunk 1]
```diff
@@ -21,7 +21,14 @@ pipeline {
                     }
                     steps {
                         sh 'ccache --zero-stats'
-                        sh 'rm -rf build && mkdir -p build && cd build && cmake -D CMAKE_INSTALL_PREFIX=$ARBORX_DIR -D CMAKE_BUILD_TYPE=Debug -D CMAKE_CXX_COMPILER_LAUNCHER=ccache -D CMAKE_CXX_COMPILER=$KOKKOS_DIR/bin/nvcc_wrapper -D CMAKE_PREFIX_PATH="$KOKKOS_DIR;$BOOST_DIR;$BENCHMARK_DIR" -D ARBORX_ENABLE_MPI=ON -D MPIEXEC_PREFLAGS="--allow-run-as-root" .. && make -j8 VERBOSE=1 && ctest --no-compress-output -T Test'
+                        sh 'rm -rf build && mkdir -p build'
+                        dir('build') {
+                            sh 'cmake -D CMAKE_INSTALL_PREFIX=$ARBORX_DIR -D CMAKE_BUILD_TYPE=Debug -D CMAKE_CXX_COMPILER_LAUNCHER=ccache -D CMAKE_CXX_COMPILER=$KOKKOS_DIR/bin/nvcc_wrapper -D CMAKE_PREFIX_PATH="$KOKKOS_DIR;$BOOST_DIR;$BENCHMARK_DIR" -D ARBORX_ENABLE_MPI=ON -D MPIEXEC_PREFLAGS="--allow-run-as-root" ..'
+                            sh 'make -j8 VERBOSE=1'
+                            timeout(time: 3, unit: 'MINUTES') {
+                                sh 'ctest --no-compress-output -T Test'
```
Comment (masterleinad) [inline .jenkins]: We could also just do 
```suggestion
                                sh 'ctest --timeout 180 --no-compress-output -T Test'
```
without the need for a `timeout` plugin.

**Comment by dalg24** [comment]
> we could also just rely on CMake functionality 

That's fine with me.  Let's see what others think.

**Comment by masterleinad** [comment]
@aprokop @Rombur Any opinion here?

**Comment by Rombur** [comment]
Don't have a real preference. I guess using `ctest` makes the jenkins file shorter...

**Comment by dalg24** [comment]
I changed.
```
$ ctest --help | grep timeout
  --test-timeout               = The time limit in seconds, internal use
  --timeout <seconds>          = Set a global timeout on all tests.
```
It is unclear to me whether that is the default per test or total time.

**Comment by dalg24** [comment]
Done

**Comment by masterleinad** [comment]
Thanks!

**Pull request merged**